### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by gomesjason**
Adds e2e tests. The Kubernetes storage e2e test suite is leveraged to test a variety of use cases. Test suites can be found here https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/testsuites

Notable points:
- Updated sanity package from v1 to v5
- Moved testing dependencies to it's own `go.mod` file. This prevents us from having to pull in packages like Kubernetes in our release.
- Uses test suites provided by Kubernetes to test a range of E2E cases. Things covered include: dynamic provisioning, static provisioning, volume expansion, snapshots. Full lifecycles are tested for all (create, mount, write, delete)
- Includes a custom test suite to test our unique use case of restoring a filesystem snapshot as a volume.

Testing:
- Ran e2e tests

### CODE REVIEW

File: tests/e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+3. Ginkgo CLI
+
+Run `ginkgo -r -p` from the tests/e2e folder to run
+
+### Empty Environment
+
+To create a testing environment the following must be preinstalled:
+1. Amazon CLI
+
+TODO
```
Comment (jacobwolfaws) [inline tests/e2e/README.md:31]: Is this supposed to still be here

**Comment by jacobwolfaws** [comment]
Can you update the PR description with some more information

File: tests/e2e/README.md (hunk 1 above)
Comment (gomesjason) [reply tests/e2e/README.md:31]: Yea, the next revision is going to include this section and bash scripts associated with it. Wanted to get initial feedback on the tests themselves first.

**Comment by gomesjason** [comment]
Just converted the PR to a draft. Right now existing PRs will change what a user can do end to end, so it'll probably break a few tests. I'll mark it as Ready for review after everything is merged and I fix any failures.

**Pull request closed without merging**